src/main/java/TaskList.java

```diff
@@ -0,0 +1,66 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents a List of tasks that the user wants to keep track of and methods to interact with the list.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Initialises a TaskList object from a file containing saved tasks
+     * @param file .txt file that contains saved Duke task data
+     * @param ui Ui to display results and errors to the user
+     */
+    @SuppressWarnings("checkstyle:MissingSwitchDefault")
+    public TaskList(File file, Ui ui) {
+        this.tasks = new ArrayList<>();
+
+        try {
+            Scanner sc = new Scanner(file);
+
+            while (sc.hasNextLine()) {
+                String[] components = sc.nextLine().split(" \\| ");
+
+                //noinspection CheckStyle
+                switch (components[0]) {
+                case "T":
+                    tasks.add(new Todo(components[2], components[1].equals("1") ? true : false));
```

Comment:
Just a minor thing, but I think you don't need the conditional expression here since .equals already returns a boolean :)